sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
+                b = base.__class__(Identity(base.shape[0]))
+                if b == Identity(base.shape[0]).as_explicit():
+                    return b
+                else:
+                    return Identity(base.shape[0])
+            except TypeError:
```

Comment:
If I don't do that, I get an assertion error for cases where we deal with Immutable matrix, for instance : 
X = ImmutableMatrix([[1, 2, 3], [4, 5, 6]])
assert MatPow(X, 1).doit() == X

can you suggest some other way out of this?